### DISCUSSION TITLE
About GPU

### DISCUSSION THREAD

**Discussion by wonwill** [YAMACS-SML/YAMACS#5] [Q&A] [answered]
Thanks to your YAMACS, I have successfully used GROMACS package.
But I have a question about GPU usage.
Is it possible that YAMACS uses GPU when my YASARA uses GPU (RTX 4090) well?
OR
Are there any additional settings to use GPU in YAMACS?
Thanks a lot.

**Comment by YAMACS-SML** [comment, marked as answer]
Hi, Thanks for using YAMACS. Yes it is possible to use GPU in YAMACS. But for that you need to modify all the YAMACS python plugins. YAMACS only works with "gmx" engine of GROMACS. If you want to run GROMACS on GPU, you need to install additional cuda driver (For more information please visit :https://manual.gromacs.org/documentation/current/user-guide/mdrun-performance.html). Unless and untill you are using "gmx" engine of GROMACS, YAMACS will work perfectly. But while using GPU, if you want to run GROMACS using MPI, then you will have to use "gmx-mpi" engine of GROAMCS. In this case you need to replace all "gmx" word with "gmx-mpi" word in all the YAMACS python plugins. That will help you to run YAMACS using GPU. If you feel any difficulties, please feel free to contact us.

**Reply by wonwill** [reply]
Thanks for your kind reply. I have one more question. 
Is it enough to replace "gmx" with "gmx_mpi" in YAMACS_3_MD.py script only? or all 5 scripts?

**Reply by YAMACS-SML** [reply]
You have to replace  "gmx" with "gmx_mpi" in all 5 plugins
